tests/config/test-sitedetailsindex.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Automattic\VIP\Config;
+
+class SiteDetailsIndex_Test extends \WP_UnitTestCase {
+	/**
+	 * Make tests run in separate processes since we're testing state
+	 * related to plugin init, including various constants.
+	 */
+	protected $preserveGlobalState = false; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.PropertyNotSnakeCase
+	protected $runTestInSeparateProcess = true; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.PropertyNotSnakeCase
+
+	public function setUp() {
+		require_once __DIR__ . '/../../config/class-sitedetailsindex.php';
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__cron_event_should_not_be_hooked_if_no_init() {
+		$sdi = new SiteDetailsIndex();
+
+		$this->assertFalse( has_filter( 'vip_site_details_index_data', [ $sdi, 'set_env_and_core' ] ) );
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__cron_event_should_be_hooked_if_init() {
+		// Getting the instance should call init which should add set_env_and_core to the hook
+		$sdi = SiteDetailsIndex::instance();
+		
+		$this->assertTrue( is_integer( has_filter( 'vip_site_details_index_data', [ $sdi, 'set_env_and_core' ] ) ) );
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__set_env_and_core() {
```

Comment:
For the ones where we can (e.g. multisite, blog_id, client_site_id, plugins), we should verify the values as well.

Reply:
`time()` (as used [here](https://github.com/Automattic/vip-go-mu-plugins/pull/1768/files?w=1#diff-60017c90bd0b751e35b1cf0a2e47709dR44)) is an external dependency (it uses the system clock). If the timestamp value was injected into the SiteDetailsIndex class, then it could be mocked and thus unit tested to verify the expected value as well. See https://matthiasnoback.nl/2018/02/mocking-at-architectural-boundaries-persistence-and-time/.

Reply:
Sure. 

Probably shouldn't do plugins since it varies depending on environment. For example I have bbpress show up when I run the tests in lando since that plugin is installed when I was testing stuff.

I added the timestamp mocking on a per-instance basis.

Reply:
Actually, scratch that on plugins. Can just mock by setting the cache.